### PR TITLE
Added character management in socket | Character data like food water…

### PR DESCRIPTION
**Pull request by Panouklakos**
… and last location will be saved in the database

I noticed the lack of character tracking for food and water in the socket managers. So I tried to implement it. I want to implement health (with the custom doctor script) as well. 

Also I edited the character manager on the socket side in order for it to track active characters, for future fetching and editing

### CODE REVIEW

File: resources/game/src/client/client.ts
Comment (spAnser) [inline resources/game/src/client/client.ts:1]: I think their server id should be part of the jwt token which is passed which should be available on all socket requests.

File: socket/src/managers/characters.ts
Comment (spAnser) [inline socket/src/managers/characters.ts:1]: The huge part with the face is very similar to https://github.com/spAnser/pioneer-village/blob/master/socket/src/controllers/characters.ts#L26

Can leave it like is, but ultimately I'll want to refactor those to a single helper function somewhere.

**Comment by spAnser** [comment]
Overall looks good. Thanks.

File: socket/src/managers/characters.ts
Comment (Panouklakos) [reply socket/src/managers/characters.ts:1]: I may as well do it in my next pull request. I want to add money and maybe job management.

**Comment by Panouklakos** [comment]
Ok I believe it is pretty much complete, any feedback I will gladly implement. Loving the project keep it going man.

**Comment by spAnser** [comment]
The only hold up right now is the jwt token should contain the server id. You should've have to pass it in with the socket message

**Comment by Panouklakos** [comment]
And that should be it

**Comment by spAnser** [comment]
I was bad and only manually reviewed it.

It seems like there is a few imports to missing files and some of the currency stuff is in the prisma schema but not a migration and is causing some query errors.

**Comment by spAnser** [comment]
The files that are imported that don't exist are:

https://github.com/spAnser/pioneer-village/blob/master/resources/base/src/client/client.ts#L6

https://github.com/spAnser/pioneer-village/blob/master/resources/ui/src/ui/app/index.tsx#L9

**Pull request merged**